### ISSUE TITLE
"CURL_IPRESOLVE_V4" bug  on hhvm

### ISSUE THREAD

**Issue by barisbora** [bugsnag/bugsnag-laravel#45]
/bugsnag/bugsnag/src/Bugsnag/notification.php line 110;

curl_setopt($http, CURL_IPRESOLVE_V4, true);

problem with php-hhvm system. If close the this option everything works well

**Comment by GrahamCampbell** [comment]
This is not a bug with this package.

**Comment by ConradIrwin** [comment]
@barisbora when you say "close this option" what do you mean?

Could we add a check that skips this line on hhvm?

**Comment by barisbora** [comment]
On hhvm, Laravel 5.0.26 with bugsnag;

i am getting this error "option 1 in not a valid",  
/bugsnag/bugsnag/src/Bugsnag/notification.php line 110;

when i was delete curl_setopt($http, CURL_IPRESOLVE_V4, true); line, problem resolved.

Note: only hhvm installed server is getting this error.

if (!defined('HHVM_VERSION')) {
     curl_setopt($http, CURL_IPRESOLVE_V4, true);
}

its worked for me. Both hhvm and normal ngix server

**Comment by snmaynard** [comment]
Would this work?

```php
if (defined('HHVM_VERSION')) {
  curl_setopt($http, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
} else {
  curl_setopt($http, CURL_IPRESOLVE_V4, true);
}
```

It looks like HHVM might just have a different [interface](http://docs.hhvm.com/manual/en/function.curl-setopt.php) to set this

**Comment by barisbora** [comment]
yes work and its better

**Comment by ConradIrwin** [comment]
Fixed in v2.5.4 of bugsnag-php. Thanks
